### ISSUE TITLE
How it works? : queryappstate -> getPIDsByName

### ISSUE THREAD

**Issue by eething** [appium/appium#16754]
* I don't know it's bug or I have missunderstanding
* I don't think code dosen't work, but it seems works? I am wondering...
* I am checking with Galaxy S21 (apilevel=31), it supports pgrep -f
```
_.escapeRegExp(`([[:blank:]]|^)${name}([[:blank:]]|$)`
```
becomes `"\(\[\[\:blank.....\$\)"`

* code is here...
  * https://github.com/appium/appium-adb/commit/ef9d8e93beaafbcfcd9ec0d8b89e98d168f87910#diff-da488f71a078a36fa4fa8b5e5b4c8cf9f441f78d018fb3014e16a9d7f36a1652L1106

**Comment by eething** [comment]
I think it's a bug (when i saw code), but i am not sure 100%...

* I tried with my phone and adb on windows
  *  adb shell pgrep -f "\\(\\[\\[...", it failed.
  * adb shell pgrep -f "\(\[\[...", it failed.
  * adb shell pgrep -f "([[:blank...", it succeeded.

**Comment by eething** [comment]
* I tested additional test before sleep... it works well... so I will CLOSE this issue
* pgrep is somewhat weird... it success and fail with or without quotes and escapes...
```
D:\>adb shell "pgrep -f '\(\[\[:blank:\]\]\|\^\)com\.android\.vending\(\[\[:blank:\]\]\|\$\)'"
//no output with QUOTES, with ESCAPES

D:\>adb shell "pgrep -f \(\[\[:blank:\]\]\|\^\)com\.android\.vending\(\[\[:blank:\]\]\|\$\)"
7036 // OK, without QUOTES, with ESCAPES

D:\>adb shell "pgrep -f '([[:blank:]]|^)com\.android\.vending([[:blank:]]|$)'"
7036 // OK, with QUOTES, without ESCAPES

D:\>adb shell "pgrep -f ([[:blank:]]|^)com\.android\.vending([[:blank:]]|$)"
/system/bin/sh: syntax error: unexpected '(' // ERROR, without QUOTES, without ESCAPES
```

**Comment by KazuCocoa** [comment]
yes, Win system had more tricky `"` `'` handling in escape, so we needed to give `windowsVerbatimArguments` for node's `exec`. i.e.  https://github.com/appium/appium-adb/pull/591/files
Potentially we have more code where require it for Win more to handle escape better
